exchange/bidder.go

```diff
@@ -392,13 +395,26 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
 			}
 		} else {
 			errs = append(errs, httpInfo.err)
+			errorCode := errortypes.ReadCode(httpInfo.err)
+			nonBidReason := ErrorToNonBidReason(errorCode)
+			seatNonBids = &openrtb_ext.SeatNonBid{
+				Seat:   string(bidderRequest.BidderName),
+				NonBid: make([]openrtb_ext.NonBid, len(bidderRequest.BidRequest.Imp)),
+			}
+			for i, imp := range bidderRequest.BidRequest.Imp {
+				// assuming same non bid reason across multiple impressions
+				nonBid := newProxyNonBid(imp.ID, int(nonBidReason))
+				nonBid.Error = httpInfo.err.Error()
+				seatNonBids.NonBid[i] = nonBid
+			}
```

Comment:
@SyntaxNode  and team:
If you look at the above code snippet, we are applying `nonBidReason` flat across all impressions here. As, we are not able to identify per impression level scenario here.